### PR TITLE
Wazuh Cloud - Cold Storage Documentation

### PR DESCRIPTION
**Pull request by sultanovich**
## Description

This PR details the process of downloading files en cold storage in Wazuh Cloud.

### Docs building
- [x] Compiles without warnings.
### Code formatting and web optimization
- [ ] Uses three spaces indentation.
- [ ] Adds or updates meta descriptions accordingly.
- [ ] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [ ] Uses present tense, active voice, and semi-formal registry.
- [ ] Uses short, simple sentences.
- [ ] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

### CODE REVIEW

**Comment by sultanovich** [comment]
Hi @JavierBejMen , this PRis ready to be applied, could you check it please?

File: source/cloud-service/archive-data/access.rst [hunk 1]
```diff
@@ -14,28 +14,32 @@ To access your archive data, you need an AWS token that grants permission on the
...
 
       curl -XPOST https://api.cloud.wazuh.com/v2/storage/token -H "x-api-key: <YOUR_API_KEY>" -H "Content-Type: application/json" --data '
       {
-         "environment_cloud_id": "0123456789ab",
+         "environment_cloud_id": "012345678ab",
          "token_expiration": "3600"
       }'
 
    .. code-block:: console
       :class: output
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:38]: Please highlight the relevant information lines with `:emphasize-lines:`, for example

```
   .. code-block:: console
      :class: output
      :emphasize-lines: 7-9

      {
         "environment_cloud_id": "012345678ab",
         "aws": {
            "s3_path": "wazuh-cloud-cold-us-east-1/012345678ab",
            "region": "us-east-1",
            "credentials": {
               "access_key_id": "mUdT2dBjlHd...Gh7Ni1yZKR5If",
               "secret_access_key": "qEzCk63a224...5aB+e4fC1BR0G",
               "session_token": "MRg3t7HIuoA...4o4BXSAcPfUD8",
               "expires_in": 3600
            }
         }
      }
```

File: source/cloud-service/archive-data/access.rst [hunk 2]
```diff
@@ -46,7 +50,12 @@ The following example describes the steps to follow to list the files of your ar
...
 
-3. Using the AWS-CLI tool to list the files, add the token to the AWS credentials file ``~/.aws/credentials``.
+
+
+Generate AWS `wazuh_cloud_storage` profile
+----------------------------------------
+
+Add the token to the AWS credentials file ``~/.aws/credentials``.
 
    .. code-block:: console
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:61]: Highlight the relevant lines with `:emphasize-lines:`.

File: source/cloud-service/archive-data/access.rst [hunk 3]
```diff
@@ -14,28 +14,32 @@ To access your archive data, you need an AWS token that grants permission on the
       See the :doc:`Wazuh Cloud CLI </cloud-service/cli/index>` section to learn how to list and download your archive data automatically.
 
 
-The following example describes the steps to follow to list the files of your archive data:
+Generate your API key
+---------------------
 
+To obtain the api key token it is necessary to execute this step described in the `Wazuh Cloud CLI </cloud-service/cli/index>` section.
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:20]:
```suggestion
Getting your API key and the AWS token
--------------------------------------

#. Obtain your Wazuh Cloud API key by following the steps outlined in the API :doc:`Authentication </cloud-service/apis/authentication>` section.
```

File: source/cloud-service/archive-data/access.rst [hunk 4]
```diff
@@ -14,28 +14,32 @@ To access your archive data, you need an AWS token that grants permission on the
...
 
-1. Before your start using the Wazuh Cloud API, you need an API key. To generate your API key, see the :ref:`Authentication <cloud_apis_auth>` section.
 
-2. Use the ``POST /storage/token`` endpoint of the :cloud-api-ref:`Wazuh Cloud API <tag/storage>` to get the AWS token and access the archive data of a specific environment. In this example, we generate an AWS token valid for 3600 seconds for environment `0123456789ab`.
+Get the AWS token
+------------------
+
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:25]:
```suggestion
```

File: source/cloud-service/archive-data/access.rst [hunk 5]
```diff
@@ -14,28 +14,32 @@ To access your archive data, you need an AWS token that grants permission on the
...
+Use the ``POST /storage/token`` endpoint of the :cloud-api-ref:`Wazuh Cloud API <tag/storage>` to get the AWS token and access the archive data of a specific environment. In this example, we generate an AWS token valid for 3600 seconds for environment `012345678ab`.
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:26]:
```suggestion
#. Use the :cloud-api-ref:`POST /storage/token <tag/storage>` API endpoint with your key to get a temporary AWS token. For example, the following request generates an AWS token valid for ``3600`` seconds that grants access to the environment archive data with ID ``012345678ab``.
```

File: source/cloud-service/archive-data/access.rst [hunk 6]
```diff
@@ -46,7 +50,12 @@ The following example describes the steps to follow to list the files of your ar
          }
       }
 
-3. Using the AWS-CLI tool to list the files, add the token to the AWS credentials file ``~/.aws/credentials``.
+
+
+Generate AWS `wazuh_cloud_storage` profile
+----------------------------------------
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:56]:
```suggestion
Generating the AWS `wazuh_cloud_storage` profile
------------------------------------------------
```

File: source/cloud-service/archive-data/access.rst [hunk 7]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
       aws_secret_access_key = qEzCk63a224...5aB+e4fC1BR0G
       aws_session_token = MRg3t7HIuoA...4o4BXSAcPfUD8
 
-4. Run the following command to list your files.
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:67]:
```suggestion
```

File: source/cloud-service/archive-data/access.rst [hunk 8]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
-      
-      $ aws --profile wazuh_cloud_storage --region us-east-1 s3 ls wazuh-cloud-cold-us-east-1/0123456789ab
+Listing archive data
+---------------------
+
+This command lists the archive data files of the environment `012345678ab`.
+
+.. code-block:: console
+
+  # aws --profile wazuh_cloud_storage --region us-east-1 s3 ls --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:75]:
```suggestion
   # aws --profile wazuh_cloud_storage --region us-east-1 s3 ls --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/
```

File: source/cloud-service/archive-data/access.rst [hunk 9]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+.. code-block:: none
+  :class: output
+
+  2024-04-19 17:50:06        493 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
+  2024-04-19 18:00:05      77759 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:81]:
```suggestion
   :class: output

   2024-04-19 17:50:06        493 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
   2024-04-19 18:00:05      77759 012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz
```

File: source/cloud-service/archive-data/access.rst [hunk 10]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+Examples
+--------
+
+Downloading archive data (multiple files)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:87]:
```suggestion
Downloading archive data – Multiple files
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: source/cloud-service/archive-data/access.rst [hunk 11]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+This command downloads in the `/home/test` directory the archive data files of the environment `012345678ab`.
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:89]:
```suggestion
This command downloads the archive data files of the environment ``012345678ab`` into the ``/home/test/`` directory.
```

File: source/cloud-service/archive-data/access.rst [hunk 12]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+.. code-block:: console
+
+  # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/ /home/test/
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:93]:
```suggestion
   # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/ /home/test/
```

File: source/cloud-service/archive-data/access.rst [hunk 13]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+.. code-block:: none
+  :class: output
+
+  download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz to output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
+  download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz to output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2100_kdBY42OvE9QJuiia.json.gz
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:99]: Use three-spaces tabulation

File: source/cloud-service/archive-data/access.rst [hunk 14]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+
+Downloading archive data (single file)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:103]:
```suggestion
Downloading archive data – Single file
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

File: source/cloud-service/archive-data/access.rst [hunk 15]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+This command downloads in the directory `/home/test` the file `012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz` from the environment `012345678ab`.
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:104]:
```suggestion
This command downloads the ``012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz`` file of the environment ``012345678ab`` into the directory ``/home/test``.
```

File: source/cloud-service/archive-data/access.rst [hunk 16]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+.. code-block:: console
+
+  # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz /home/test/
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:108]:
```suggestion
   # aws --profile wazuh_cloud_storage --region us-east-1 s3 cp --recursive s3://wazuh-cloud-cold-us-east-1/012345678ab/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz /home/test/
```

File: source/cloud-service/archive-data/access.rst [hunk 17]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
+
+.. code-block:: none
+  :class: output
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:111]:
```suggestion
   :class: output
```

File: source/cloud-service/archive-data/access.rst [hunk 18]
```diff
@@ -55,10 +64,50 @@ The following example describes the steps to follow to list the files of your ar
...
 
-You now have access to your archive data.
+  download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz to ./012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
```
Comment (javimed) [inline source/cloud-service/archive-data/access.rst:113]:
```suggestion
   download: s3://wazuh-cloud-cold-us-east-1/012345678ab/output/alerts/2024/04/19/012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz to ./012345678ab_output_alerts_20240419T2050_VqaWCpX9oPfDkRpD.json.gz
```

**Review by javimed — CHANGES_REQUESTED** [review]

**Review by javimed — APPROVED** [review]

**Comment by javimed** [comment]
Added commit and merged changes on demand from @dariommr

**Pull request merged**